jax/_src/cache_key.py

```diff
@@ -129,27 +132,42 @@ def _log_cache_key_hash(hash_obj, last_serialized: str, hashfn):
     )
 
 
-def _serialize_ir(m: ir.Module) -> bytes:
+def _serialize_ir(m: ir.Module, remove_custom_partitioning_ptr_for_cache_key: bool) -> bytes:
   output = io.BytesIO()
   m.operation.write_bytecode(file=output)
-  return output.getvalue()
-
-
-def _canonicalize_ir(m_original: ir.Module) -> bytes:
+  
+  str_m = str(m)
+  m_bytecode = output.getvalue()
+
+  if remove_custom_partitioning_ptr_for_cache_key and "CustomSPMDPartitioning" in str_m:
+    pattern = r'backend_config\s*=\s*"([^"]*)"'
+    matches = re.findall(pattern, str_m)
+    for match in matches:
+      b_backend_config = match.encode('utf-8')
+      b_zeros = ('0' * len(b_backend_config)).encode('utf-8')
+      backend_config_idx = m_bytecode.find(b_backend_config)
```

Comment:
Doing a find isn't great. This is a slow implementation in many hit. Do the match tell this information? 
I'd not don't do a full find from the start of the string. Start it from the previous one to keep the implementation linear in the string size.